java/ql/test/TestUtilities/JavaInlineExpectations.ql

```diff
@@ -0,0 +1,88 @@
+import java
+
```

Comment:
Should we require a `@kind post-processing` here and add some checks for the absence or presence of query predicates?

Reply:
Note that the CLI does check for absence or presence of query predicates: https://github.com/github/semmle-code/blob/a7be21bddf8376d14d87c2a5c4e2207c5279b36c/odasa-buildutils/src/com/semmle/cli2/test/CodeqlTestCase.java#L400-L413

That being said, I could add an additional check for `@kind post-processing`.

Reply:
Note that there wouldn't really be a relationship between that metadata and the check for absence or presence of query predicates. The check is due to the query being _used_ to postprocess query results (i.e. referred to with `postprocess: ` in a .qlref). If a `@kind post-processing` was ever used in a different way (e.g. referred to with `query: ` in a .qlref), that would not do any checks but just give an error.

Reply:
I have gone ahead and added a commit each to this PR and https://github.com/github/semmle-code/pull/50493 that together introduce `@kind post-processing` and the corresponding checks.

Reply:
Great. Then there's a chance that the checks can run in the IDE when editing the post-processing query (although I don't know whether that's wired up yet).